react/components/SKUSelector/components/SKUSelector.js

```diff
@@ -10,18 +11,20 @@ export default class SKUSelector extends Component {
   render() {
     const { selectedId, onSelectSku, mainVariation, secondaryVariation, maxSkuPrice } = this.props
 
+    const onSkuSelected = curry(onSelectSku) 
```

Comment:
Just a doubt, what `curry` does? The ramda doc isnt enough to understand it. :)

Reply:
it makes you able to _partially_ apply a function, meaning you can call it with some arguments, and it returns another function with those arguments set, e.g., if you have a function that receives 4 arguments and you call it with 2, it returns another function that only need the other two to be actually executed.

Reply:
All of Ramda's functions are _curried_ be default, and it is actually a very nice feature. It kind of "prepare" a function that has more than one argument. It is one of the main reason that Ramda functions' have their parameters kind of "reversed" comparing with our imperative programming view.

If I have `const sum = (a, b) => a + b`, I can get a function `sumWith3` with `const sumWith3 = curry(sum)(3)`. It **prepares** a function for a future apply.

If, for example, I need to map over a collection and sum all elements with 3, I would use my brand new function.

Some functional languages like F# have curry built-in in their core, and every function you create is already curried.

[Read more here](https://stackoverflow.com/questions/36314/what-is-currying)

Reply:
Nice! Very thanks guys, indeed  is a powerful tool .